eland/operations.py

```diff
@@ -195,12 +195,54 @@ def aggs(self, query_compiler, pd_aggs, numeric_only=None) -> pd.DataFrame:
             results, index=pd_aggs, dtype=(np.float64 if numeric_only else None)
         )
 
+    def mode(
+        self,
+        query_compiler: "QueryCompiler",
+        pd_aggs: List[str],
+        is_dataframe: bool,
+        es_size: int,
+        numeric_only: bool = False,
+        dropna: bool = True,
+    ) -> Union[pd.DataFrame, pd.Series]:
+
+        results = self._metric_aggs(
+            query_compiler,
+            pd_aggs=pd_aggs,
+            numeric_only=numeric_only,
+            dropna=dropna,
+            es_mode_size=es_size,
+        )
+
+        pd_dict: Dict[str, Any] = {}
+        row_diff: Optional[int] = None
+
+        if is_dataframe:
+            # If multiple values of mode is returned for a particular column
+            # find the maximum length and use that to fill dataframe with NaN/NaT
+            rows_len = max([len(value) for value in results.values()])
+            for key, values in results.items():
+                row_diff = rows_len - len(values)
+                # Convert np.ndarray to list
+                values = list(values)
+                if row_diff:
+                    if isinstance(values[0], pd.Timestamp):
```

Comment:
If we call `.mode()` on a dataframe that has no entries for a field but entries in other fields should we look at the es_type on the field instead of assuming if the value is a `pd.Timestamp` then we put `NaT`?

Reply:
We should always look at `field.nan_value`. But at this point of code we just return a dictionary that contains results.
I should re-fetch the fields using `fields = query_compiler._mappings.all_source_fields()`

and then if we have a row difference then it should look `values.extend([field.nan_value] * row_diff)`.

This should eliminate checking if it is timestamp or not.

@sethmlarson Shall I make it this way?

Reply:
Eh, this is fine for now, however I wonder if what we're seeing here indicates that instead of returning a dictionary of {str: result} for results we should return a dictionary of {Field: result}? Can be done later.